src/console/plugin.ts

```diff
@@ -237,3 +264,58 @@ function activateConsole(app: JupyterLab, services: IServiceManager, rendermime:
 
   mainMenu.addMenu(menu, {rank: 50});
 }
+
+
+/**
+ * A namespace for private data.
+ */
+namespace Private {
+  /**
+   * An interface for caption options.
+   */
+  export
+  interface ICaptionOptions {
+    /**
+     * The time when the console was opened.
+     */
+    opened: Date;
+
+    /**
+     * The time when the console last executed its prompt.
+     */
+    executed?: Date;
+
+    /**
+     * The path of the console file on the server.
+     */
+    path: string;
+
+    /**
+     * The label of the console (as displayed in tabs).
+     */
+    label: string;
+
+    /**
+     * The display name of the console's kernel.
+     */
+    displayName: string;
+  }
+
+  /**
+   * Generate a caption for a console's title.
+   */
+  export
+  function caption(options: ICaptionOptions): string {
+    let { label, path, displayName, opened, executed } = options;
+    let caption = (
+      `Name: ${label}\n` +
+      `Path: ${path}\n` +
```

Comment:
If I already have `pathTracker.path`, does `dirname` add anything else that I need?